src/server/worker/workerrequests.js

```diff
@@ -291,14 +291,9 @@ function WorkerRequests(mainLogger, gmeConfig, webgmeUrl) {
                         pluginManager.runPluginMain(plugin, finish);
                     })
                     .catch(function (err) {
-                        var pluginResult = pluginManager.getPluginErrorResult(plugin.id,
-                            pluginName,
-                            'Exception was raised, err: ' + err.stack,
-                            plugin && plugin.projectId);
-
-                        pluginManager.logger.error(err.stack);
-                        finish(err.message, pluginResult);
-                    });
+                        finish(err.message);
```

Comment:
Now you're returning the promise, so it should be part of the same "promise-chain" and only one catch should be needed.